### PR TITLE
Fix resident IDP config tests order

### PR DESCRIPTION
**Pull request by GANGANI**
### Proposed changes in this pull request
This PR will fix the issue of the order of the resident IDP config tests. These integration tests are to test resident idp inbound authentication config URLs and inbound provisioning config URLs in super tenant legacy URL mode and tenant legacy URL mode.

Related PR: https://github.com/wso2/product-is/pull/8370

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java [hunk 1]
```diff
@@ -40,230 +44,209 @@
...
+        this.userPassword = context.getContextTenant().getTenantAdmin().getPassword();
+        this.activeTenant = context.getContextTenant().getDomain();
     }
 
-    @Test(groups = "wso2.is", description = "Test resident IdP config URLs in super tenant mode")
-    public void testResidentIdPConfigs() throws Exception {
+    @BeforeClass(alwaysRun = true)
+    public void testInit() throws Exception {
 
-        log.info("Retrieving resident identity provide");
```
Comment (pamodaaw) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java:56]: Isn't it good to have a log to track the status of the tests execution?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java [hunk 2]
```diff
@@ -40,230 +44,209 @@
...
+                        "Expected Token Revocation Endpoint URL is not found"},
+                {"carbon.super", "OIDCLogoutEPUrl", "https://localhost:9853/oidc/logout",
+                        "Expected Logout Endpoint URL not found"},
+                {"carbon.super", "OAuth2AuthzEPUrl", "https://localhost:9853/oauth2/authorize",
+                        "Expected Authorization Endpoint URL is not found"},
+                {"carbon.super", "OAuth2DCREPUrl", "https://localhost:9853/api/identity/oauth2/dcr/v1.1/register",
+                        "Expected Dynamic Client Registration Endpoint URL is not found"},
+                {"carbon.super", "OIDCDiscoveryEPUrl", "https://localhost:9853/oauth2/oidcdiscovery",
+                        "Expected Dynamic Client Registration Endpoint URL is not found"},
+                {"carbon.super", "OAuth2JWKSPage", "https://localhost:9853/oauth2/jwks", "Expected is not found"},
```
Comment (pamodaaw) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java:129]: Shall we mention what is not found here as well?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java [hunk 3]
```diff
@@ -40,230 +44,209 @@
...
+                {"wso2.com", "OIDCLogoutEPUrl", "https://localhost:9853/oidc/logout",
+                        "Expected Logout Endpoint URL not found in the tenant mode"},
+                {"wso2.com", "OAuth2AuthzEPUrl", "https://localhost:9853/oauth2/authorize",
+                        "Expected Authorization Endpoint URL is not found in the tenant mode"},
+                {"wso2.com", "OAuth2DCREPUrl", "https://localhost:9853/t/wso2.com/api/identity/oauth2/dcr/v1.1/register",
+                        "Expected Dynamic Client Registration Endpoint URL is not found in the tenant mode"},
+                {"wso2.com", "OIDCDiscoveryEPUrl", "https://localhost:9853/t/wso2.com/oauth2/oidcdiscovery",
+                        "Expected Dynamic Client Registration Endpoint URL is not found in the tenant mode"},
+                {"wso2.com", "OAuth2JWKSPage", "https://localhost:9853/t/wso2.com/oauth2/jwks",
+                        "Expected is not found in the tenant mode"},
```
Comment (pamodaaw) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java:171]: Shall we mention what is not found here as well?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java [hunk 4]
```diff
@@ -40,230 +44,209 @@
...
-        for (FederatedAuthenticatorConfig config : authConfigs) {
-            for (Property property: config.getProperties()) {
-                fedAuthConfigMap.put(property.getName(), property.getValue());
-            }
-        }
+    @DataProvider(name = "federatedAuthConfigURLProvider")
+    public static Object[][] federatedAuthConfigURLProvider() {
+        return new Object[][]{
+                {"carbon.super", "OAuth1AccessTokenUrl", "https://localhost:9853/oauth/access-token",
+                        "Expected OAuth10a Access Token URL is not found "},
```
Comment (pamodaaw) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java:92]: Unnecessary space at the end

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java (hunk 1 above)
Comment (GANGANI) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java:56]: Fixed https://github.com/wso2/product-is/pull/9463/commits/40f51782b1b30ff3722701becd32b4020dcf84b8

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java (hunk 2 above)
Comment (GANGANI) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java:129]: Fixed 40f5178

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java (hunk 4 above)
Comment (GANGANI) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java:92]: Fixed 40f5178

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java (hunk 3 above)
Comment (GANGANI) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java:171]: Fixed 40f5178

**Pull request merged**